src/pip/_internal/wheel_builder.py

```diff
@@ -86,11 +86,7 @@ def _should_build(
 
     if not is_wheel_installed():
         # we don't build legacy requirements if wheel is not installed
-        logger.info(
-            "Using legacy 'setup.py install' for %s, "
-            "since package 'wheel' is not installed.",
-            req.name,
-        )
+        req.legacy_install_reason = 8559
```

Comment:
Can we have an enum or something for this? Or maybe even a concrete class so we can simply do

```python
if self.legacy_install_reason is not None:
    self.legacy_install_reason.emit()  # Calls deprecate() internally
```

Reply:
I'm not sure this will help much because depending on the reason, we have two different places to report the error. The other is:

https://github.com/pypa/pip/blob/82c23d9198510e4079b4e2d1cd6c39144bf343de/src/pip/_internal/req/req_install.py#L839-L850

Reply:
In total there will be 3 different reasons (the third will be `--no-binary`).

Reply:
@uranusjr I implemented your proposal because there may actually be a 4th reason (the use of `--install-option` or `--global-option`). Thanks for suggesting it!